src/js/components/Box/StyledBox.js

```diff
@@ -206,6 +206,23 @@ const gapStyle = (directionProp, gap, responsive, wrap, theme) => {
   const responsiveMetric = responsive && breakpoint && breakpoint.edgeSize[gap];
 
   const styles = [];
+  if (typeof gap === 'object') {
+    if (gap.row !== undefined && gap.column !== undefined) {
+      styles.push(
+        `gap: ${theme.global.edgeSize[gap.row] || gap.row} ${
+          theme.global.edgeSize[gap.column] || gap.column
+        };`,
+      );
+    }
+    if (gap.row !== undefined && gap.column === undefined) {
```

Comment:
can we simplify the logic a bit by making these `else if` statements and removing the second check?
```
if (gap.row !== undefined && gap.column !== undefined) { ... }
else if (gap.row !== undefined) { ... }
else if (gap.column !== undefined) { ... }
```